envoy/type/percent.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package envoy.type;
+
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Percent]
+
+// Identifies a percentage, in the range [0.0, 100.0].
+message Percent {
+  double value = 1 [(validate.rules).double = {gte: 0, lte: 100}];
+}
+
+// A fractional percentage is used in cases in which for performance reasons performing floating
+// point to integer conversions during randomness calculations is undesirable. The message includes
+// both a numerator and denominator that together determine the final fractional value.
+// * Example: 1/100 = 1%.
+// * Example: 3/10000 = 0.03%.
+message FractionalPercent {
+  // Specifies the numerator. Defaults to 0.
+  uint32 numerator = 1;
+
+  // Specifies the denominator. If not specified, defaults to 100. If the denominator
+  // specified is less than the numerator, the final fractional percentage is capped at 1 (100%).
+  google.protobuf.UInt32Value denominator = 2 [(validate.rules).uint32.gt = 0];
```

Comment:
Do we really support arbitrary denominator, such as 1000000001.  For basic types like this, the spec must be super tight. The current spec seems to allow things that would be accurately supported.

Reply:
How about I make it an enum with values of 100, 10,000, and 1,000,000 to start? Then the default can be 100 (in enum form) which is pretty nice. I will update. I agree the arbitrary denominator is not really needed.